examples/kvSourceGoPlugin.md

```diff
@@ -0,0 +1,234 @@
+[Secrets]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#secret-v1-core
+[ConfigMaps]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#configmap-v1-core
+[base64]: https://tools.ietf.org/html/rfc4648#section-4
+[Go plugin]: https://golang.org/pkg/plugin
+
+# Generating Secrets
+
+## What's a Secret?
+
+Kubernetes [ConfigMaps] and [Secrets] are
+both key:value maps, but the latter is
+intended to signal that its values have a
+sensitive nature - e.g. ssh keys or
+passwords.
+
+Kubernetes assumes that the values in a
+Secret are [base64] encoded, and decodes them
+before actual use (as, say, the argument to a
+container command).  The user that creates
+the Secret must base64 encode the data, or
+use a tool that does it for them.  This
+encoding doesn't protect the secret from
+anything other than an over-the-shoulder
+glance.
+
+Protecting the actual secrecy of a Secret
+value is up to the cluster operator. They
+must lock down the cluster (and its `etcd`
+data store) as tightly as they feel
+necessary, and likewise protect the bytes
+that feed into the cluster to ultimately
+become the content of a Secret value.
+
+## Make a place to work
+
+<!-- @establishBase @test -->
+```
+DEMO_HOME=$(mktemp -d)
+```
+
+## Secret values from local files
+
+kustomize has three different ways to generate a secret
+from local files:
+
+ * get raw values from the kustomization file itself,
+ * get them from so-called _env_ files (`NAME=VALUE`, one per line),
+ * consume the entire contents of a file to make one secret value.
+ 
+Here's an example combining all three methods:
+
+Make an env file with some short secrets:
+
+<!-- @makeEnvFile @test -->
+```
+cat <<'EOF' >$DEMO_HOME/foo.env
+ROUTER_PASSWORD=admin
+DB_PASSWORD=iloveyou
+EOF
+```
+
+Make a text file with a long secret:
+
+<!-- @makeLongSecretFile @test -->
+```
+cat <<'EOF' >$DEMO_HOME/longsecret.txt
+Lorem ipsum dolor sit amet,
+consectetur adipiscing elit,
+sed do eiusmod tempor incididunt
+ut labore et dolore magna aliqua.
+EOF
+```
+
+And make a kustomization file referring to the
+above and additionally defining some KV pairs
+literally in the file:
+
+<!-- @makeKustomization1 @test -->
+```
+cat <<'EOF' >$DEMO_HOME/kustomization.yaml
+secretGenerator:
+- name: mysecrets
+  kvSources:
+  - pluginType: builtin
+    name: literals
+    args:
+    - FRUIT=apple
+    - VEGETABLE=carrot
+  - pluginType: builtin
+    name: files
+    args:
+    - longsecret.txt
+  - pluginType: builtin
+    name: envfiles
+    args:
+    - foo.env
```

Comment:
This format is different from what's supported in 2.0.3. Considering users of kubectl(vendoring kustomize 2.0.3) will also look at this doc, can we use the 2.0.3 supported format?

Reply:
Good point - done.